core/src/main/scala/pureconfig/ConfigCursor.scala

```diff
@@ -66,7 +66,33 @@ sealed trait ConfigCursor {
    *         otherwise.
    */
   def asListCursor: Either[ConfigReaderFailures, ConfigListCursor] =
-    castOrFail(LIST, Set(LIST), _.asInstanceOf[ConfigList]).right.map(ConfigListCursor(_, pathElems))
+    if (isUndefined) {
+      failed(KeyNotFound.forKeys(path, Set()))
+    } else {
+      val ve = value.valueType() match {
+        case ConfigValueType.OBJECT =>
```

Comment:
This subtly changes the behavior of `asCollectionCursor` - it will now yield `Left` for array-like objects!

Reply:
If we do the `DefaultTransformer` transformations under the hood, we don't need an `asCollectionCursor` config anymore. That said, I'd prefer to keep the method deprecated in our API if we can implement the old behavior here.